theano/sandbox/multinomial.py

```diff
@@ -193,6 +193,78 @@ def perform(self, node, ins, outs):
                             z[0][n, m] += 1
                             break
 
+class WeightedSelectionFromUniform(Op):
+    """
+    Converts samples from a uniform into sample from a multinomial.
+
+    """
+
+    __props__ = ("odtype",)
+
+    def __init__(self, odtype):
+        self.odtype = odtype
+
+    def __str__(self):
+        return '%s{%s}' % (self.__class__.__name__, self.odtype)
+
+    def __setstate__(self, dct):
+        self.__dict__.update(dct)
+        try:
+            self.odtype
+        except AttributeError:
+            self.odtype = 'auto'
```

Comment:
This try/except is not needed. All it does is ensure that the Op has an attribute `odtype` which is guaranteed by the fact that the `__init__()` method created that attribute and there has never been a version of this op without that attribute (since this op is new from your PR)